telemetry_core/src/main/java/com/newrelic/telemetry/spans/Span.java

```diff
@@ -249,7 +249,35 @@ public SpanBuilder parentId(String parentId) {
      * @return The SpanBuilder instance with the error field set to true
      */
     public SpanBuilder withError() {
+      return withError(null, null);
+    }
+
+    /**
+     * Call this to indicate that the span contains an error condition with the given message.
+     *
+     * @param errorMessage The error message to be placed into the "error.message" attribute
+     * @return The SpanBuilder instance with the error field set to true
+     */
+    public SpanBuilder withError(String errorMessage) {
+      return withError(errorMessage, null);
+    }
+
+    /**
+     * Call this to indicate that the span contains an error condition with the given message and
+     * class.
+     *
+     * @param errorMessage The error message to be placed into the "error.message" attribute
+     * @param errorClass The error class to be placed into the "error.class" attribute
+     * @return The SpanBuilder instance with the error field set to true
+     */
+    public SpanBuilder withError(String errorMessage, String errorClass) {
       this.error = true;
+      if (errorMessage != null) {
+        attributes.put("error.message", errorMessage);
```

Comment:
this is the first time we've had the builder manipulate the attributes. If we want to go down this route, we should probably 
a) deprecate the `attributes` bulk-adding method and
b) add a method to add an attribute directly.

In any case, we need to document in both of these new methods that if you call `attributes` after calling these, that you will lose what you just set.

Reply:
Agreed that it's the first time that this builder adds attributes, but why do you think we should deprecate the bulk add?  The single add seems useful, but we haven't had an immediate need yet?  

I can definitely add a blurb about overwriting.

Reply:
I was just thinking that having 2 conflicting things was a bit confusing. I'm ok with it, if you think documentation is good enough.

Reply:
What do our exporters do today? Would they be simpler if they were able to add the attributes directly to the builder, rather than having to build the attributes, then set them?

Reply:
the OT one builds them together and passes them in https://github.com/newrelic/opentelemetry-exporter-java/blob/f65f3451261ba2bfcf66909b4bc9f524ac06537e/opentelemetry-exporters-newrelic/src/main/java/com/newrelic/telemetry/opentelemetry/export/SpanBatchAdapter.java#L64

Reply:
But would it be simpler just to use the SpanBuilder for that purpose? I don't know, honestly...just thought it would be a good question to ask.

Reply:
I'm ok with this if you are.  It seems in line with what was originally suggested and I don't find it offensive.  :)